### ISSUE TITLE
ActionView::Template::Error ( is not a recognized provider)

### ISSUE THREAD

**Issue by Overload119** [carrierwaveuploader/carrierwave#1939]
```
    carrierwave (0.11.0)
      activemodel (>= 3.2.0)
      activesupport (>= 3.2.0)
      json (>= 1.7)
      mime-types (>= 1.16)
```

`carrierwave.rb`

```
CarrierWave.configure do |config|
  config.fog_credentials = {
    :provider               => 'AWS',       # required
    :aws_access_key_id      => Rails.application.secrets.aws_access_key_id,
    :aws_secret_access_key  => Rails.application.secrets.aws_secret_access_key,
  }
  config.fog_directory = 'instant-pages'
  config.use_action_status = true
end
```

**Comment by Overload119** [comment]
This issue seemed to fix itself.. I have no idea how.
I added a `pry` statement in the view, to see where the issue was from 

L11 `fog-core-1.36.0/lib/fog/core/services_mixin.rb` 

But it just started working...

**Comment by Overload119** [comment]
Just a little more detail for anyone else (because I ran into this issue again)

1. I added a `pry` statement to see the source of the issue.
2. My uploader class `MyUploader.fog_credentials` was returning `{}` hence no fog credentials.

When I start rails, this runs properly:

`gems/fog-core-1.36.0/lib/fog/core/services_mixin.rb` with the argument `attributes` being set as the configuration in my initializer, ie. `CarrierWave.configure do |config| ...`

`ProfileImageUploader.storage` returns `CarrierWave::Storage::Fog`

`include CarrierWaveDirect::Uploader` was included in `ProfileImageUploader` and it shouldn't have been, since that clears the fog credentials for some reason.

Anyway, now I know why that happened so hopefully this will help someone else.
